packages/orchestration/src/exos/chainAccountKit.js

```diff
@@ -0,0 +1,200 @@
+          );
+        },
+        getLocalAddress() {
+          return NonNullish(
+            this.state.localAddress,
+            'local address not available',
+          );
+        },
+        getRemoteAddress() {
+          return NonNullish(
+            this.state.remoteAddress,
+            'remote address not available',
+          );
+        },
+        getPort() {
+          return this.state.port;
+        },
+        executeTx() {
+          throw new Error('not yet implemented');
+        },
+        /**
+         * Submit a transaction on behalf of the remote account for execution on the remote chain.
+         * @param {AnyJson[]} msgs
+         * @param {Omit<TxBody, 'messages'>} [opts]
+         * @returns {Promise<string>} - base64 encoded bytes string. Can be decoded using the corresponding `Msg*Response` object.
+         * @throws {Error} if packet fails to send or an error is returned
+         */
+        executeEncodedTx(msgs, opts) {
+          const { connection } = this.state;
+          if (!connection) throw Fail`connection not available`;
+          return E.when(
+            E(connection).send(makeTxPacket(msgs, opts)),
+            // if parsePacketAck cannot find a `result` key, it throws
+            ack => parsePacketAck(ack),
+          );
+        },
+        /**
+         * Close the remote account
+         */
+        async close() {
+          /// XXX what should the behavior be here? and `onClose`?
+          // - retrieve assets?
+          // - revoke the port?
+          const { connection } = this.state;
+          if (!connection) throw Fail`connection not available`;
+          await E(connection).close();
+        },
+        async deposit(payment) {
+          console.log('deposit got', payment);
+          throw new Error('not yet implemented');
+        },
+        /**
+         * get Purse for a brand to .withdraw() a Payment from the account
+         * @param {Brand} brand
+         */
+        async getPurse(brand) {
+          console.log('getPurse got', brand);
+          throw new Error('not yet implemented');
+        },
+
+        /* transfer account to new holder */
+        async prepareTransfer() {
+          throw new Error('not yet implemented');
+        },
+      },
+      connectionHandler: {
+        /**
+         * @param {Remote<Connection>} connection
+         * @param {LocalIbcAddress} localAddr
+         * @param {RemoteIbcAddress} remoteAddr
+         */
+        async onOpen(connection, localAddr, remoteAddr) {
+          trace(`ICA Channel Opened for ${localAddr} at ${remoteAddr}`);
+          this.state.connection = connection;
+          this.state.remoteAddress = remoteAddr;
+          this.state.localAddress = localAddr;
+          // XXX parseAddress currently throws, should it return '' instead?
+          this.state.chainAddress = harden({
+            address: parseAddress(remoteAddr) || UNPARSABLE_CHAIN_ADDRESS,
```

Comment:
The name `parseAddress` suggests that it's taking something like `agoric123lj23kj2...` and picking it apart. I had to read the source code -- well, the tests, actually -- to figure out what it's doing. How about changing its argument type to `RemoteIbcAddress`? and maybe rename it to `findAddressField`?

p.s. I sure wish we used tools that make it easier to re-use tests as documentation a la [doctest](https://en.wikipedia.org/wiki/Doctest).

Reply:
Good naming suggestion. Will tackle it here if there's add'l feedback, or tackle it as part of #9198.

Also planning to rename `makeICAConnectionAddress` to `makeICAChannelAddress`